### PR TITLE
Unskip IIIF rotation test

### PR DESCRIPTION
**Pull request by rcantin-w**
# Unskip IIIF rotation test

- For https://github.com/wellcomecollection/wellcomecollection.org/issues/13109

## What does this change?

Donald said their work was done (see ticket for details), so we should be ok to unskip this test.

While working on this, we discovered and fixed two bugs related to image rotation:

**Bug 1: Rotation not preserved when resizing**  
When an image failed to load due to IIIF server size restrictions, the error handler would fetch a smaller version but was not preserving the rotation parameter from the original URL, causing rotated images to reset to 0°.

**Bug 2: State not resetting between rotations**  
The `tryLoadingSmallerImg` state in `IIIFViewerImage.tsx` was never resetting between image changes. This meant that if an image errored and fell back to a smaller size on the first rotation, subsequent rotations back to that same angle would skip the size fallback and display a broken image.

These bugs were only visible in Playwright automated testing:

> [!NOTE]
> The rotation bug was more visible in Playwright than manual browser testing because Playwright runs with a clean browser context (no cached images), makes rapid automated clicks, and likely triggers the IIIF server's size limit checks more consistently. In regular Chrome browsing, cached images and slower human interaction patterns mean the error path is less frequently triggered, making the rotation preservation bug harder to spot.

Here's me recording it in the Playwright environment with manual testing. First rotation doesn't work (shows the smaller version with lost rotation) and the fifth one then shows the broken image.

https://github.com/user-attachments/assets/840300b2-0560-436b-9508-6b88661eeddd

### Changes made:

1. **Unskipped the rotation test** in `playwright/test/view-items.test.ts`
2. **Improved the test assertion** to use a regex pattern with `toHaveAttribute`, which provides auto-retry behaviour but still checks the URL contains `/90/default.jpg`
3. **Fixed rotation preservation bug** in `common/utils/convert-image-uri.ts`:
   - The `convertIiifImageUri` function now extracts the rotation parameter from the original IIIF URL
   - When resizing images (e.g., due to size limit errors), the rotation (90°, 180°, 270°) is preserved in the new URL
   - Added detailed comments explaining the IIIF Image API URL structure
4. **Fixed state reset bug** in `content/webapp/views/pages/works/work/IIIFViewer/IIIFViewerImage.tsx`:
   - Added a `useEffect` hook that resets both `tryLoadingSmallerImg` and `hasLoaded` whenever the `src` prop changes
   - Ensures each new image gets a fresh attempt at the size conversion fallback if needed
5. **Added unit tests** in `content/webapp/test/utils/convert-image-uri.test.ts` to verify:
   - Rotation is preserved when resizing images
   - Rotation of 0 degrees is handled correctly
   - Malformed URLs without rotation default gracefully
   - Rotation persists through multiple conversion operations

## How to test

All tests should pass.

## Have we considered potential risks?

The Playwright test could start failing again down the line if there are issues with the IIIF service, but we can always re-skip it if needed.

The rotation preservation and state reset fixes are defensive - they only apply rotation when present in the original URL, handle edge cases (missing/invalid values) gracefully, and follow standard React patterns for resetting state on prop changes.

### CODE REVIEW

File: playwright/test/view-items.test.ts [hunk 1]
```diff
@@ -159,9 +159,7 @@ test('(8) | Licence information should be available', async ({
   await expect(page.getByText('Credit:')).toBeVisible();
 });
 
-// Please unskip once this is fixed by Digirati
-// https://wellcome.slack.com/archives/CBT40CMKQ/p1779797736981459
-test.skip('(9) | The image should rotate', async ({ page, context }) => {
+test('(9) | The image should rotate', async ({ page, context }) => {
   await itemWithSearchAndStructures(context, page);
   await page.getByRole('button', { name: 'Rotate' }).click();
   const currentIndex = await page.getByTestId('active-index').textContent();
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply playwright/test/view-items.test.ts:165]: Sounds like a good idea, I'll change it.

File: common/utils/convert-image-uri.ts [hunk 2]
```diff
@@ -115,10 +115,20 @@ export function convertIiifImageUri(
   } else {
     const imageIdentifier = originalUri.split(iiifImageUri)[1].split('/', 2)[0];
 
+    // Extract rotation from original URL if present
+    // IIIF Image API format: {identifier}/{region}/{size}/{rotation}/{quality}.{format}
+    // https://iiif.io/api/image/3.0/#21-image-request-uri-syntax
+    // For well-formed IIIF URLs, after splitting by base URL and then by '/':
+    //   urlParts[0] = identifier, [1] = region, [2] = size, [3] = rotation, [4] = quality.format
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply common/utils/convert-image-uri.ts:122]: I hadn't pushed...

**Review by gestchild — APPROVED** [review]
